pkgs/native_assets_cli/lib/native_assets_cli_internal.dart

```diff
@@ -8,13 +8,23 @@
 /// @nodoc
 library native_assets_cli_internal;
```

Comment:
What is the purpose of this library. Who is intended to use it?
Consider putting it inside `src/`, maybe `src/internal_use/`, then whoever imports it has to use an `//ignore:` to avoid a warning, but that's reasonable.

Reply:
It's used by `native_assets_builder`, `dartdev`, and `flutter_tools`.

That's a good suggestion. However it does mean we don't get alerted on API changes to it by https://pub.dev/packages/dart_apitool. So I guess the question is whether the versioning of this package should just concern `build.dart` writers and `native_assets_builder`, `dartdev`, and `flutter_tools` should be on their own. Maybe they should.

Reply:
Do consider documenting who this is intended for.


```dart
/// Internal API for building assets.
///
/// This library is intended for use by:
///  * `package:native_assets_builder`,
///  * `dartdev` (in the Dart SDK), and,
///  * `flutter_tools` (in the Flutter SDK).
///
/// > [!CAUTION]
/// > Unless you are building a custom SDK that embeds Dart you should
/// > not be importing this library!
///
/// @nodoc
library native_assets_cli_internal;
```

That way it's also clear why it's here for future maintainers.

Also if `> [!CAUTION]` is new to you, it's really cool -- check out:
https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts

It's supported by recent versions of `dartdoc` and coming to a pub.dev server near you soon!